Open Judge System/Services/OJS.Services/Business/Contests/ContestsImportService.cs

```diff
@@ -0,0 +1,187 @@
+namespace OJS.Services.Business.Contests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using OJS.Data;
+    using OJS.Data.Models;
+    using OJS.Services.Business.Contests.Models;
+
+    public class ContestsImportService : IContestsImportService
+    {
+        private readonly IOjsDbContext dbContext;
+
+        public ContestsImportService(IOjsDbContext dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+
+        public ContestImportResult ImportContest(Contest contest, int categoryIdToImportTo, bool replace)
+        {
+            var contestImportedFromId = contest.Id;
+            var categoryImportedFromId = contest.CategoryId;
+            var existingContest = this.dbContext.Contests
+                .SingleOrDefault(c => c.CategoryId == categoryIdToImportTo && c.Name == contest.Name);
+
+            replace = replace && existingContest != null;
+
+            var problems = contest.ProblemGroups.SelectMany(pg => pg.Problems).ToList();
+
+            this.TryAddExistingCheckers(problems);
+            this.TryAddExistingSubmissionTypes(problems);
+
+            contest.CategoryId = categoryIdToImportTo;
+            contest.ImportedOn = DateTime.Now;
+
+            if (replace)
+            {
+                // TODO: Implement replacing
```

Comment:
Will this be implemented later on?

Reply:
Yes. probably.. but for now is only copying.